numpy/core/shape_base.pyi

```diff
@@ -2,7 +2,7 @@ from collections.abc import Sequence
 from typing import TypeVar, overload, Any, SupportsIndex
 
 from numpy import generic
-from numpy._typing import ArrayLike, NDArray, _ArrayLike
+from numpy._typing import ArrayLike, NDArray, _ArrayLike, _CastingKind
```

Comment:
```suggestion
from numpy import _CastingKind
```
`_CastingKind` is currently stored in the main namespace (really should move it at some point...)